addons/stock_account/models/product.py

```diff
@@ -8,17 +8,48 @@
 from collections import defaultdict
 from datetime import datetime
 
+from odoo.addons.stock_account import const
+
 
 class ProductTemplate(models.Model):
     _inherit = ['product.template']
 
-    cost_method = fields.Selection(related="categ_id.property_cost_method", readonly=True)
-    valuation = fields.Selection(related="categ_id.property_valuation", readonly=True)
+    cost_method = fields.Selection(
+        string="Cost Method",
+        selection=const.COST_METHOD,
+        compute='_compute_cost_method',
+    )
+    valuation = fields.Selection(
+        string="Valuation",
+        selection=const.VALUATION,
+        compute='_compute_valuation',
+    )
     lot_valuated = fields.Boolean(
         "Valuation by Lot/Serial number",
         help="If checked, the valuation will be specific by Lot/Serial number.",
     )
 
+    @api.depends("categ_id.property_cost_method")
+    def _compute_cost_method(self):
+        for product_template in self:
+            product_template.cost_method = (
+                product_template.categ_id.property_cost_method
```

Comment:
```suggestion
                product_template.categ_id.with_company(
                    product_template.company_id
                ).property_cost_method
```
If the product is assigned to a specific company, we should make sure the property is read for the company of the product.